java/clients/src/main/java/sleeper/clients/status/update/DeleteTable.java

```diff
@@ -59,8 +59,8 @@ public DeleteTable(InstanceProperties instanceProperties, AmazonS3 s3Client, Tab
 
     public void delete(String tableName) throws StateStoreException {
         TableProperties tableProperties = tablePropertiesStore.loadByName(tableName);
-        deleteAllObjectsInBucketWithPrefix(s3Client, instanceProperties.get(DATA_BUCKET), tableProperties.get(TABLE_ID));
         stateStoreProvider.getStateStore(tableProperties).clearSleeperTable();
+        deleteAllObjectsInBucketWithPrefix(s3Client, instanceProperties.get(DATA_BUCKET), tableProperties.get(TABLE_ID));
         tablePropertiesStore.deleteByName(tableName);
```

Comment:
Can we add a comment to explain the logic to the order in which these calls are made?

Reply:
Thanks, added javadoc explaining the reasoning of the order of operations

Reply:
I don't think it belongs in the Javadoc because it's only relevant when you're reading the implementation code.